File: src/main/java/com/pubnub/api/core/models/PNStatus.java
package com.pubnub.api.core.models;

import com.pubnub.api.core.enums.PNStatusCategory;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PNStatus {

    PNStatusCategory category;
    boolean error;
    boolean automaticallyRetry;

    public void retry(){
        // TODO
    }

    public void cancelAutomaticRetry() {
        // TODO
    }


}

File: pubnub-kotlin-core/src/main/kotlin/com/pubnub/api/models/consumer/PNStatus.kt
package com.pubnub.api.models.consumer

import com.pubnub.api.PubNubException
import com.pubnub.api.enums.PNStatusCategory

data class PNStatus(
    val category: PNStatusCategory,
    val exception: PubNubException? = null,
    val currentTimetoken: Long? = null,
    val channels: Collection<String> = emptySet(),
    val channelGroups: Collection<String> = emptySet()
) {
    @get:JvmName("isError")
    val error: Boolean = exception != null
}
